### PR TITLE
feat(request): add timeout and retry logic

### PR DESCRIPTION
**Pull request by giangbui**
Description about what this pull request does.

Add timeout and retry logic for all requests made to indexd api

Please make sure to follow the [DEV guidelines](https://gen3.org/resources/developer/dev-introduction/) before asking for review.

### New Features
Add timeout and retry logic for all requests made to indexd api

### CODE REVIEW

File: indexclient/client.py [hunk 1]
```diff
@@ -31,6 +31,20 @@ def handle_error(resp):
             resp.raise_for_status()
 
 
+def indexd_request_wrapper(func):
+    def retry_logic_with_timeout(*args, **kwargs):
+        kwargs.update({"timeout": 60})
```
Comment (fantix) [inline indexclient/client.py]: did you mean to use `set_default()`? The `update` here overwrites any given `timeout`.

File: indexclient/client.py [hunk 2]
```diff
@@ -305,21 +311,25 @@ def list_versions(self, did):
...
         return versions
 
+    @indexd_request_wrapper
     def _get(self, *path, **kwargs):
         resp = requests.get(self.url_for(*path), **kwargs)
         handle_error(resp)
         return resp
 
+    @indexd_request_wrapper
     def _post(self, *path, **kwargs):
```
Comment (fantix) [inline indexclient/client.py]: I'm still having concerns adding retries on `POST`, `PUT` or `DELETE` - it is totally fine to add timeouts, but retrying means any future IndexD APIs must be strictly repeatable (calling with the same args for more than one times ends up with identical results), if current APIs are already repeatable. What was the initial motivation to retry on e.g. `POST` please?

File: indexclient/client.py (hunk 1 above)
Comment (giangbui) [reply indexclient/client.py]: Good catch. Thanks!

File: indexclient/client.py (hunk 2 above)
Comment (giangbui) [reply indexclient/client.py]: I have a ticket to copy and index million files. We have to make sure that all the new files need to be indexed in indexd. For safety, I think I will only add `timeout` for `POS`, `PUT` and `DELETE` and change the logic in my code. Thank you for your comment!
Fixed in 5f39c8bcb7a47e5052033c6ad2b7d0e3df0f5b72

**Pull request merged**